### DISCUSSION TITLE
Inconsistent battery charging behavior

### DISCUSSION THREAD

**Discussion by vladyspavlov** [vladyspavlov/esphome-must-inverter#12] [General]
Inverter model is PV18-5248 Pro.

When configuring the inverter with a Li (or USE) battery type, the charging behavior does not follow the expected absorption and float pattern. Instead, the inverter cycles between bulk and float voltage, resulting in frequent micro-cycling.

I'm using utility as the primary charging source, the inverter stays at or returns to bulk voltage, not dropping to the intended resting float level.

I don't have solar panels for now, so I can't confirm this behavior in the PV mode, but as users report on the forums, there is no issue in the PV mode.

Also tried to connect/disconnect the battery CAN communication cable, it doesn't have an effect.

Inverter settings:
14 Battery type - Li (also tried USE)
17 Absorb voltage - 57.2 V
18 Float voltage - 53.8 V
37 BMS control method - VOL

<img src="https://github.com/user-attachments/assets/3d2e6e0c-3a33-47d7-acfe-164693a86723" width=600>

More details and comments here:
https://powerforum.co.za/topic/19603-must-inverter-settings/?do=findComment&comment=179012

**Comment by LionBS76** [comment]
can't open image when click on it

UPD:  strange but already can

**Comment by LionBS76** [comment]
Inverter PV18-3224 (24 V model)

Have set
(14) Bat type - USE
(17) Bulk voltage - 28 V
(18) Float voltage - 27 V

As I could understand different MUST model - different behaviour
Inverter charge battery to Bulk voltage and then stop charging.
Battery voltage drop to float voltage and inverter start to maintain this voltage.


![image](https://github.com/user-attachments/assets/66b7be5d-18c9-4798-9488-46ada9e7bfd7)
